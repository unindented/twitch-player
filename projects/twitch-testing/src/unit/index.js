import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
