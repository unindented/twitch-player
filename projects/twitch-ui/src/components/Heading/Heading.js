import PropTypes from "prop-types";
import React from "react";
import { StyleSheet } from "react-native";
import Text from "../Text";
import { useTheme } from "../../hooks";

const Heading = ({ level, style = [], children, testID = "heading" }) => {
  const { colors, typography } = useTheme();
  const color = colors.heading;
  const fontSize =
    level === "1"
      ? typography.sizePrimary
      : level === "2"
      ? typography.sizeSecondary
      : typography.sizeTertiary;
  const compositeStyle = [styles.root]
    .concat(style)
    .concat({ color, fontSize });

  return (
    <Text
      accessibilityRole="heading"
      aria-level={level}
      style={compositeStyle}
      testID={testID}
    >
      {children}
    </Text>
  );
};

Heading.propTypes = {
  level: PropTypes.oneOf(["1", "2", "3"]).isRequired,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.arrayOf(PropTypes.object),
  ]),
  children: PropTypes.node.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    fontWeight: "bold",
  },
});

export default Heading;
