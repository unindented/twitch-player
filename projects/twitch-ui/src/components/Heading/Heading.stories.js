import { storiesOf } from "@storybook/react";
import React from "react";
import Heading from ".";

storiesOf("Heading", module)
  .add("Level 1", () => <Heading level="1">Heading Level 1</Heading>)
  .add("Level 2", () => <Heading level="2">Heading Level 2</Heading>)
  .add("Level 3", () => <Heading level="3">Heading Level 3</Heading>);
