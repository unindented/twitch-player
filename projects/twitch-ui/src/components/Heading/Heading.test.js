import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import Heading from "./Heading";

describe("Heading", () => {
  let instance;

  describe("level 1", () => {
    beforeEach(() => {
      instance = render(<Heading level="1">Some Heading</Heading>);
    });

    it("renders a level-1 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });

  describe("level 2", () => {
    beforeEach(() => {
      instance = render(<Heading level="2">Some Heading</Heading>);
    });

    it("renders a level-2 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });

  describe("level 3", () => {
    beforeEach(() => {
      instance = render(<Heading level="3">Some Heading</Heading>);
    });

    it("renders a level-3 heading", () => {
      expect(instance.getByTestId("heading")).toMatchSnapshot();
    });
  });
});
