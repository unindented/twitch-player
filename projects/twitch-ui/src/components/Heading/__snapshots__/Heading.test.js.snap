// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading level 1 renders a level-1 heading 1`] = `
<h1
  aria-level="1"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 32px;"
>
  Some Heading
</h1>
`;

exports[`Heading level 2 renders a level-2 heading 1`] = `
<h2
  aria-level="2"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 16px;"
>
  Some Heading
</h2>
`;

exports[`Heading level 3 renders a level-3 heading 1`] = `
<h3
  aria-level="3"
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-testid="heading"
  dir="auto"
  role="heading"
  style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 13px;"
>
  Some Heading
</h3>
`;
